services/issue/template.go

```diff
@@ -109,21 +109,23 @@ func IsTemplateConfig(path string) bool {
 	return false
 }
 
-// GetTemplatesFromDefaultBranch checks for issue templates in the repo's default branch,
-// returns valid templates and the errors of invalid template files.
-func GetTemplatesFromDefaultBranch(repo *repo.Repository, gitRepo *git.Repository) ([]*api.IssueTemplate, map[string]error) {
-	var issueTemplates []*api.IssueTemplate
-
+// ParseTemplatesFromDefaultBranch parses the issue templates in the repo's default branch,
+// returns valid templates and the errors of invalid template files (the errors map is guaranteed to be non-nil).
+func ParseTemplatesFromDefaultBranch(repo *repo.Repository, gitRepo *git.Repository) (ret struct {
+	IssueTemplates []*api.IssueTemplate
+	TemplateErrors map[string]error
+},
```

Comment:
I think I'd rather define a concrete struct type above, even if not exported.
I don't love named returns, much less an _anonymous_ named return.

Reply:
> I think I'd rather define a concrete struct type above, even if not exported.

That's impossible to pass the lint, it violates "do not export private names"

>  I don't love named returns, much less an anonymous named return.

But I do love this usage. Anonymous struct is perfect use case for such function.

Reply:
This is nearly the same thing, I'd rather nolint this case than use an anonymous struct, personally.

Reply:
Sorry but I didn't get the point why "anonymous struct" is bad, we use it a lot.

The code is readable enough and there should be no maintainability problem, and no one needs to "grep" the name.